sktime/transformations/compose/_column.py

```diff
@@ -145,6 +185,42 @@ def _transformers(self, value):
                 )
             ]
 
+    def _check_transformers(self, X):
+        transformers = self.transformers
+
+        if isinstance(transformers, BaseTransformer):
+            for col in X.columns:
+                transformers = [(str(col), transformers, pd.Index([col]))]
+
+        coerced_trafo = []
+        indices = []
+        for name, transformer, index in transformers:
+            c_index = self._coerce_to_pd_index(index, X)
+            indices += [c_index]
+            coerced_trafo += [(name, transformer, c_index)]
+        transformers = coerced_trafo
+
+        # handle remainder
+        remainder = self.remainder
+
+        if remainder == "passthrough" or isinstance(remainder, BaseTransformer):
+            if isinstance(remainder, BaseTransformer):
+                rem_t = self.remainder.clone()
+            elif remainder == "passthrough":
+                from sktime.transformations.compose import Id
+
+                rem_t = Id()
```

Comment:
To reduce the code complexity, here's a non-blocking suggestion:

1. After L204, check if remainder is `drop` or not. If yes, return.
2. Rest of the logic can go unindented skipping L206 check.

Reply:
yes, changed to that.

Reply:
Personally I'm in favour of positive checks when possible, and what I meant is this:

```python
if remainder == "drop" or remainder is None:
```

But this is equivalent I suppose (please let me know if that's not the case), so I've no further comments.

```python
if remainder != "passthrough" and not isinstance(remainder, BaseTransformer):
```

Reply:
welllll, my reasoning in favour of the negative check is to maximize the scope of the default behaviour in case of an input check failure.

Suppose in some future scenario where the input check in `__init__` gets removed or breaks accidentally.

Then the default `"drop"` happens only for a positive condition that you have to specifically hit.

Whereas, currently, the default happens whenever you do not hit the narrow negative condition, which makes the default more robust to accidents or changes.

Reply:
(but, assuming the input check in `__init__` has passed and nothing modifies `self.remainder`, the two conditions are afaik indeed equivalent)